jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+source "${JBOSS_HOME}/bin/launch/launch-common.sh"
+source "${JBOSS_HOME}/bin/launch/jboss-kie-common.sh"
+source "${JBOSS_HOME}/bin/launch/logging.sh"
+
+# make sure there is a workbench service within the namespace.
+serviceName=${WORKBENCH_SERVICE_NAME/-/_}
+serviceHost=$(find_env "${serviceName^^}_SERVICE_HOST")
+servicePort=$(find_env "${serviceName^^}_SERVICE_PORT_HTTP")
+
+# param
+# ${1} - the config map payload
+update_config_map() {
+    local payload=${1}
+    # only execute the following lines if this container is running on OpenShift
+    if [ -e /var/run/secrets/kubernetes.io/serviceaccount/token ]; then
+        local namespace=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+        local token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+        local response=$(curl -s --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+            -H "Authorization: Bearer $token" \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/strategic-merge-patch+json' \
+            -XPATCH \
+            https://${KUBERNETES_SERVICE_HOST:-kubernetes.default.svc}:${KUBERNETES_SERVICE_PORT:-443}/api/v1/namespaces/${namespace}/configmaps/myapp-kieserver -d ${payload} )
+    fi
+}
+
+if [ "${serviceHost}x" != "x" -a "${servicePort}x" != "x" ]; then
+    auth="$(echo -n "${KIE_ADMIN_USER}:${KIE_ADMIN_PWD}" | base64)"
+    if [ -n "${KIE_SERVER_ID}" ]; then
+        uri="deploymentconfigs?labelSelector=services.server.kie.org%2Fkie-server-id%3D${KIE_SERVER_ID}"
+        response=$(query_ocp_api "apis/apps.openshift.io" "${uri}")
+        replicas=$(echo ${response:: -3} | python -c 'import json,sys;obj=json.load(sys.stdin);print (obj["items"][0]["spec"]["replicas"])')
+
+        bcResponse=$(query_ocp_api "apis/apps.openshift.io" "deploymentconfigs/${WORKBENCH_SERVICE_NAME}")
+        bcReplicas=$(echo ${bcResponse:: -3} | python -c 'import json,sys;obj=json.load(sys.stdin);print (obj["status"]["readyReplicas"])')
+
+        if [ ${replicas} == 0 ]; then
+            log_info "KIE Server Replicas is ${replicas}, notifying Business Central controller."
+            curl -s -X DELETE -i "http://${serviceHost}:${servicePort}/rest/controller/server/${KIE_SERVER_ID}" -H "Content-Type: application/json" -H "Authorization: Basic ${auth}" \
+            && \
+            update_config_map "{\"metadata\":{\"labels\":{\"services.server.kie.org/kie-server-state\":\"DETACHED\"}}}"
+
+        elif [ ${replicas} -gt 0 -a ${bcReplicas} -gt 0 ]; then
```

Comment:
How about change the logic to something like this: (In pseudo code)
```
if ( kieDC.spec.replicas >  0)  {
   // postStart hook
   update_config_map -> USED
   if ( workbenchServiceIsDefined && bcDC.spec.replicas > 0) {
     /**
      * curl command may be hanging a bit during dc pod starting up; need to update 
      * Pod.spec.terminationGracePeriodSeconds so as to safe guard infinite wait 
      * and also accommodate bc Pod startup time.
      */
     curl -s -X PUT ....
   } 
}
```

Reply:
agreed.